wallet/wallet-cli-commands/src/command_handler/mod.rs

```diff
@@ -154,6 +164,63 @@ where
                         whether_to_store_seed_phrase.to_bool(),
                         mnemonic,
                         passphrase,
+                        hardware_wallet,
+                    )
+                    .await?;
+
+                self.wallet.update_wallet::<N>().await;
+
+                let msg = match newly_generated_mnemonic.mnemonic {
+                    MnemonicInfo::NewlyGenerated {
+                        mnemonic,
+                        passphrase,
+                    } => {
+                        let passphrase = if let Some(passphrase) = passphrase {
+                            format!("passphrase: {passphrase}\n")
+                        } else {
+                            String::new()
+                        };
+                        format!(
+                            "New wallet created successfully\nYour mnemonic: {}\n{passphrase}\
+                        Please write it somewhere safe to be able to restore your wallet. \
+                        It's recommended that you attempt to recover the wallet now as practice\
+                        to check that you arrive at the same addresses, \
+                        to ensure that you have done everything correctly.
+                        ",
+                            mnemonic
+                        )
+                    }
+                    MnemonicInfo::UserProvided => "New wallet created successfully".to_owned(),
+                };
+
+                Ok(ConsoleCommand::SetStatus {
+                    status: self.repl_status().await?,
+                    print_message: msg,
+                })
+            }
+
+            WalletManagementCommand::RecoverWallet {
+                wallet_path,
+                mnemonic,
+                whether_to_store_seed_phrase,
+                passphrase,
+                hardware_wallet,
+            } => {
+                let hardware_wallet = hardware_wallet.and_then(|t| match t {
+                    #[cfg(feature = "trezor")]
+                    CliHardwareWalletType::Trezor => Some(HardwareWalletType::Trezor),
+                    CliHardwareWalletType::None => None,
+                });
+
+                let newly_generated_mnemonic = self
```

Comment:
This doesn't seem right. When the wallet is recovered from Software mnemonic is provided by user. In Hardware it's not provided but I don't think that HW ever exposes it. So it seems like NewlyGenerated block should be unreachable.